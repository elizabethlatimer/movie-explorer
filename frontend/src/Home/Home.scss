.Home {
    /*Background image of homepage*/
    img {
        /* Set rules to fill background */
        min-height: 100%;
        min-width: 1024px;

        /* Set up proportionate scaling */
        width: 100%;
        height: auto;

        /* Set up positioning */
        position: fixed;
        top: 0;
        left: 0;
        z-index: -1;
    }

    .HomeSearch {
        width: 60%;
        margin: auto;
        margin-top: 15%;
    }
}
