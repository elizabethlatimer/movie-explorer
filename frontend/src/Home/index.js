import React, {useState} from 'react';
import { InputGroup, Button, FormControl } from 'react-bootstrap';

import './Home.scss';
import background from '../shared/images/background-home.jpg'


function Home({search}) {
  const [formData, setFormData] = useState({search: ''});

  function handleChange(evt) {
    let { name, value } = evt.target;
    setFormData(formData => ({ ...formData, [name]: value }));
  }

  return (
    <div className="Home">
      <img src={background}
        alt="books lying open layered across floor" />
      <div className="HomeSearch">
        <InputGroup className="mb-3">
          <FormControl
            placeholder="Find a Movie"
            aria-label="Find a Movie"
            aria-describedby="basic-addon2"
            value={formData.search}
            onChange={handleChange}
            name="search"
          />
          <InputGroup.Append>
            <Button variant="primary">Search</Button>
          </InputGroup.Append>
        </InputGroup>
      </div>
      Photo by Jeremy Yap on Unsplash
    </div>
  )
}

export default Home;