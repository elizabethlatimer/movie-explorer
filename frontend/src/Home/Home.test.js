import React from 'react';
import { render } from "@testing-library/react";

import Home from './index';
import MovieContext from '../shared/helpers/movieContext'

const values = { setMovieList: () => jest.fn(), setCurrentQuery: () => jest.fn() };

test('renders without crashing', () => {
  render(
    <MovieContext.Provider value={values}>
      <Home />
    </MovieContext.Provider>);
});

test('matches the snapshot', () => {
  const { asFragment } = render(<MovieContext.Provider value={values}>
    <Home />
  </MovieContext.Provider>);
  expect(asFragment()).toMatchSnapshot();
});