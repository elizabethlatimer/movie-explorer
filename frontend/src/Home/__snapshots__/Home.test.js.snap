// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <img
      alt="movie projector shining light"
      src="background-home.jpg"
    />
    <div
      class="HomeSearch"
    >
      <form
        class=""
      >
        <div
          class="mb-3 input-group"
        >
          <input
            aria-describedby="basic-addon2"
            aria-label="Find a Movie"
            class="form-control"
            name="search"
            placeholder="Find a Movie"
            value=""
          />
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="photoAttribution"
    >
      <p>
        Background Photo by Jeremy Yap on Unsplash
      </p>
      <div>
        <img
          alt="The Movie Database Logo"
          class="TMDBlogo"
          src="https://www.themoviedb.org/assets/2/v4/logos/v2/blue_square_2-d537fb228cf3ded904ef09b136fe3fec72548ebc1fea3fbbd1ad9e36364db38b.svg"
        />
         This app uses the TMDb API but is not endorsed or certified by TMDb.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
